File: java/ContactsApp/app/src/main/java/com/egehanasal/contactsapp/ContactAdapter.java
package com.egehanasal.contactsapp;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.egehanasal.contactsapp.databinding.RecyclerRowBinding;

import java.util.ArrayList;

public class ContactAdapter extends RecyclerView.Adapter<ContactAdapter.ContactHolder> {

    ArrayList<Contact> contactArrayList;

    public ContactAdapter(ArrayList<Contact> contactArrayList) {
        this.contactArrayList = contactArrayList;
    }

    @NonNull
    @Override
    public ContactHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        RecyclerRowBinding recyclerRowBinding = RecyclerRowBinding.inflate(LayoutInflater.from(parent.getContext()),parent,false);
        return new ContactHolder(recyclerRowBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull ContactHolder holder, int position) {
        String s = contactArrayList.get(position).getName() + " " + contactArrayList.get(position).getSurname();
        holder.binding.recyclerViewTextView.setText(s);

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(holder.itemView.getContext(), ContactInfoActivity.class);
                Singleton singleton = Singleton.getInstance();
                singleton.setSentContact(contactArrayList.get(position));
                holder.itemView.getContext().startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
        return contactArrayList.size();
    }

    public class ContactHolder extends RecyclerView.ViewHolder{

        private RecyclerRowBinding binding;

        public ContactHolder(RecyclerRowBinding binding){
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: kotlin/ContactsApp/app/src/main/java/com/egehanasal/contactsapp/ContactAdapter.kt
package com.egehanasal.contactsapp

import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.egehanasal.contactsapp.databinding.RecyclerRowBinding

class ContactAdapter (private val contactList: ArrayList<Contact>) : RecyclerView.Adapter <ContactAdapter.ContactHolder>() {

    private lateinit var recyclerViewString: String
    class ContactHolder (val binding: RecyclerRowBinding) : RecyclerView.ViewHolder(binding.root) {

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ContactHolder {
        val binding = RecyclerRowBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ContactHolder(binding)
    }

    override fun onBindViewHolder(holder: ContactHolder, position: Int) {
        recyclerViewString = contactList.get(position).name + " " + contactList.get(position).surname
        // contactList.get(position) yerine contactList[position] da yazılabiliyor.
        holder.binding.recyclerRowTextView.text = recyclerViewString

        holder.itemView.setOnClickListener {
            val intent = Intent(holder.itemView.context, DetailsActivity :: class.java)
            intent.putExtra("contact", contactList[position])
            holder.itemView.context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return contactList.size
    }
}